network/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,184 @@
+
+### Polling and Retrying Transactions
+
+#### Async Transaction Submission
+
+Submissions using the `/transactions_async` endpoint return an immediate response back from Stellar-Core. There are different actions that clients can take based on the specific `tx_status` returned:
+
+1. `PENDING`: The submission is successful but the transaction is still waiting to be included in a ledger. You should use the [GET /transactions/:transaction_hash](/network/horizon/api-reference/resources/retrieve-a-transaction) endpoint to poll for the submitted transaction and check if it makes into a ledger.
+2. `ERROR`: The submission did not go through due to an error in Stellar-Core. Take a look at the attached error message for more details, modify your transaction if necessary, and resubmit.
+3. `TRY_AGAIN_LATER`: This indicates that the Stellar-Core instance is currently unable to process submission of this particular transaction. Clients should wait for sometime before resubmitting the transaction. This could happen due to different reasons:
+    - There is another transaction from same source account in memory
+    - It has been rejected due to too low fee and has been resubmitted too soon
+    - The transaction has resource consumption higher than the network limits allowClients should wait for sometime before retrying it again.
+
+#### Synchronous Transaction Submission
+
+Due to the blocking nature of this endpoint, things are a little different compared to the asynchronous strategy. There are 3 possible scenarios that clients can encounter:
+
+1. The submission is successful and Horizon returns the transaction response back. This is the happy path and clients do not need to do anything but wait for Horizon's response.
+2. Stellar-Core sends back an `ERROR` response from the submission. Clients should consult the attached error message and retry the submission again.
+3. Timeouts: Horizon may respond with a `504` HTTP code. This response is not an error but a warning that your transaction hasn't been accepted by the network yet. There could be many possible reasons for the timeout, the most common of which is network congestion, but it could also be due to other transient issues.
+   1. **Polling the transaction hash**: Use the transaction hash in the timeout response and poll the [GET /transactions/:transaction_hash](/network/horizon/api-reference/resources/retrieve-a-transaction) endpoint to see if it successfully makes it into a ledger.
+   2. **Timebounds**: Timebounds are optional but **highly** recommended as they put a definitive time limit on the transaction's finality - after it times out, you will know for sure whether it made it into a ledger.
+   3. **Resubmitting the transaction**: If the transaction status poll has not returned anything or if your transaction has timed out based on the time bounds you specified, you should attempt a resubmission of the same transaction.
```

Comment:
@Shaptic correct me if I'm wrong, but I don't think clients should resubmit a transaction before it expires in any case, right? So the phrase "if the transaction status poll has not returned anything" should be removed?

If you get a 404 after polling for the transaction _after_ the transaction has expired, that means Horizon is lagging behind, in which case you shouldn't resubmit and you have to keep polling until it catches back up. If Horizon isn't lagging, then you won't get a 404 after the transaction expires, you'll get a 400 with a tx_too_late error code.

Reply:
Yeah that's a good observation. I mean, you _can_ keep resubmitting and it'll only get in once, but that's only because of Horizon's helpfulness and so it's not an ideal understanding of the txsub flow. 

In general, "if your transaction has timed out based on the time bounds you specified, you should attempt a resubmission of the same transaction," this will always fail unless you update the timebounds. Better is to, as you said, wait until timebounds, poll until you get a status, then rebuild (extend) and submit again. Also "has not returned anything" is ambiguous; there's always something returned unless you're talking about a 5xx.